### DISCUSSION TITLE
Start with hy on Anaconda

### DISCUSSION THREAD

**Discussion by GiorgosCham** [hylang/hy#2418] [General]
Hi all,
I am new to github and to this particular forum. I've installed via anaconda 'hy' and now would like to start coding. How do I open the editor environment?
GC

**Comment by Kodiologist** [comment]
I have only a foggy idea what Anaconda is. You'd better look at Anaconda's documentation or ask the Anaconda people.

**Reply by GiorgosCham** [reply]
how do you open the editor if there's no anaconda installed?

**Reply by Kodiologist** [reply]
Me? I use Emacs, which I launch with the command `emacs`. Hy is just a programming language; it doesn't include an editor. You can use whatever editor you like. We have some not-quite-maintained packages to provide extra Hy-related features for [Emacs](https://github.com/hylang/hy-mode) and [Vim](https://github.com/hylang/vim-hy), and there's a shiny new add-on for [Visual Studio Code](https://github.com/hylang/vscode-hy), but these are just conveniences; all you actually need is a text editor.

**Reply by GiorgosCham** [reply]
u run that command within a python cell ?

**Reply by Kodiologist** [reply]
`emacs`? No, I run that from the system shell. Jupyter is not a shell.

If you're new to programming and you're running Windows, I would suggest installing [Notepad++](https://notepad-plus-plus.org) and using that while you're learning Hy. Learning how to use Emacs or vim or some other fancy editor is a big task best left for later.

**Reply by GiorgosCham** [reply]
yep I run windows but still when I open a command prompt and enter commands from the tutorials they arent recognized..

**Reply by Kodiologist** [reply]
`cmd.exe` is not a Bourne-compatible shell, nor does it interpret Hy code. It has its own syntax descended from DOS.

**Reply by GiorgosCham** [reply]
ok, well I paste commands from 1st tutorial in python cell and wont work although I've done the pip process with anaconda and cmd 
![εικόνα](https://user-images.githubusercontent.com/126391925/221412699-d95dc846-5247-46aa-b2b1-e55164d9b8aa.png)

**Reply by Kodiologist** [reply]
A Jupyter Python cell only supports Python. Hy is a different language.

**Reply by Kodiologist** [reply]
[This message is a reply to a deleted message.]

From the Hy REPL, or from a text file that you execute with Hy.

**Reply by GiorgosCham** [reply]
so notepad ++ can be read from hy?

**Reply by Kodiologist** [reply]
No, but a text file that you write and save with Notepad++ can.

**Reply by GiorgosCham** [reply]
ok, so how do I open up a Hy REPL?

**Reply by Kodiologist** [reply]
With the command `hy`. From `cmd.exe`, not from Jupyter.

**Reply by GiorgosCham** [reply]
![εικόνα](https://user-images.githubusercontent.com/126391925/221413193-90d20c56-3b1d-415e-b539-0d763f0923e6.png)

**Reply by GiorgosCham** [reply]
Thats probably a better pic than the previous one

![εικόνα](https://user-images.githubusercontent.com/126391925/221413352-770693fa-c91c-42eb-859b-43e30623c9ce.png)

**Reply by Kodiologist** [reply]
You should copy and paste error messages and example code, rather than taking screenshots.

It looks like the package is installed but `hy` isn't in your  PATH. This is not a Windows help forum, so at this point I would ask you to try working on this yourself. Remember that there's also a web-based Hy REPL at https://hylang.org/try-hy.

**Reply by GiorgosCham** [reply]
nice..thx a bunch!

**Comment by goyalyashpal** [comment]
> A Jupyter Python cell only supports Python. Hy is a different language.
\- @ Kodiologist at https://github.com/hylang/hy/discussions/2418#discussioncomment-5114194

u massively mixed concepts there lol

corrections:
* an ipython cell only supports python
* to use `hy` in a jupyter notebook: install `hy` and the hy kernel for jupyter

hy also has a jupyter kernel - i don't know since when: https://github.com/jupyter/jupyter/wiki/Jupyter-kernels

----
to add `hy` to path in windows 10: 

1. search for "environment variables" or "environment variables for your account" in the windows search bar (start menu)
2. in the top half part of the opened window "Environment Variables" : scroll to find "Path" -> select it, click edit 
3. in the "Edit environment variable" window : click new > Paste the path _to the folder_ containing `hy` executable.
4. Click Ok, Ok

To verify it's done properly: 
1. Run a fresh `cmd` instance 
2. execute `hy` 
3. or alternatively execute `env` and search/eyeball for "PATH=" and thereafter eyeball for your added path.

**Comment by goyalyashpal** [comment]
> Thats probably a better pic than the previous one
> \- @ GiorgosCham at https://github.com/hylang/hy/discussions/2418#discussioncomment-5114252
> 
> ![image](https://user-images.githubusercontent.com/126391925/221413352-770693fa-c91c-42eb-859b-43e30623c9ce.png)

i am facing the same; and on trying to inspect the directory structure: there is no executable anywhere:

1. not in the `Python\Python311\Scripts\` where it should be, and
2. obviously not in this directory `Python\Python311\Lib\site-packages\hy\`

so, i think this is an actual issue - "executable not supplied for `pip install`s on windows"
_Addendum:_ this was addressed in this issue: https://github.com/hylang/hy/issues/2460#issuecomment-1607865385

**Reply by Kodiologist** [reply]
If you think there's a bug, then go ahead and file an issue, but you should copy and paste error messages and example code, rather than taking screenshots.

**Reply by goyalyashpal** [reply]
there's no example code here 😅
